Common/Securities/Option/OptionFilterUniverse.cs

```diff
@@ -220,27 +220,68 @@ public OptionFilterUniverse Strikes(int minStrike, int maxStrike)
             _isDynamic = true;
 
             // find the current price in the list of strikes
+            var exactPriceFound = true;
             var index = _uniqueStrikes.BinarySearch(_underlying.Price);
+
+            // Return value of BinarySearch (from MSDN):
+            // The zero-based index of item in the sorted List<T>, if item is found;
+            // otherwise, a negative number that is the bitwise complement of the index of the next element that is larger than item
+            // or, if there is no larger element, the bitwise complement of Count.
             if (index < 0)
             {
                 // exact price not found
+                exactPriceFound = false;
+
+                if (index == ~_uniqueStrikes.Count)
+                {
+                    // there is no greater price, return empty
+                    _allSymbols = Enumerable.Empty<Symbol>();
+                    return this;
+                }
+
                 index = ~index;
 
                 if (index >= _uniqueStrikes.Count)
                 {
-                    // price out of range: should never happen, return empty
+                    // should never happen, return empty
```

Comment:
anytime I see 'should never happen' I think we should throw an exception or _minimally_ log as much state as possible